utils/post_processing/restructure_borealis.py

```diff
@@ -0,0 +1,507 @@
+# Copyright 2019 SuperDARN Canada, University of Saskatchewan
+# Author: Liam Graham
+
+#
+# restructure_borealis.py
+# 2019-08-08
+
+import deepdish as dd
+import numpy as np
+import sys
+import os
+import subprocess as sp
+import datetime
+import warnings
+import tempfile
+
+
+########################## RESTRUCTURING CODE #################################
+# Functions for restructuring Borealis files
+# for bfiq, antennas iq, and raw acf data into a smaller,
+# faster, and more usable format
+
+shared_antiq = ['antenna_arrays_order', 'beam_azms', 'beam_nums', 'borealis_git_hash', 'data_descriptors',
+				'data_normalization_factor', 'experiment_comment', 'experiment_id', 'experiment_name',
+				'freq', 'intf_antenna_count', 'main_antenna_count', 'num_samps', 'num_slices',
+				'pulse_phase_offset', 'pulses', 'rx_sample_rate', 'samples_data_type', 'scan_start_marker',
+				'slice_comment', 'station', 'tau_spacing', 'tx_pulse_len']
+
+shared_bfiq =  ['antenna_arrays_order', 'beam_azms', 'beam_nums', 'blanked_samples', 'borealis_git_hash', 
+				'data_descriptors', 'data_normalization_factor', 'experiment_comment', 'experiment_id',
+				'experiment_name', 'first_range', 'first_range_rtt', 'freq', 'intf_antenna_count', 'lags',
+				'main_antenna_count', 'num_ranges', 'num_samps', 'num_slices', 'pulse_phase_offset', 'pulses',
+				'range_sep', 'rx_sample_rate', 'samples_data_type', 'scan_start_marker', 'slice_comment',
+				'station', 'tau_spacing', 'tx_pulse_len']
+
+shared_rawacf = ['beam_azms', 'beam_nums', 'blanked_samples', 'borealis_git_hash', 'correlation_descriptors',
+				 'data_normalization_factor', 'experiment_comment', 'experiment_id', 'experiment_name',
+				 'first_range', 'first_range_rtt', 'freq', 'intf_antenna_count', 'lags', 'main_antenna_count',
+				 'num_slices', 'pulses', 'range_sep', 'rx_sample_rate', 'samples_data_type', 'scan_start_marker',
+				 'slice_comment', 'station', 'tau_spacing', 'tx_pulse_len', 'correlation_dimensions']
+
+def find_max_sequences(data):
+	"""
+	Finds the maximum number of sequences between records in a Borealis data file
+	Args
+		data:		Site formatted data from a Borealis file, 
+					organized as one record for each slice
+	Returns:
+		max_seqs:	The largest number of sequences found 
+					in one record from the file
+	"""
+	first_key = list(data.keys())[0]
+	max_seqs = data[first_key]["num_sequences"]
+	for k in data:
+		if max_seqs < data[k]["num_sequences"]:
+			max_seqs = data[k]["num_sequences"]
+	return max_seqs
+
+
+def antennas_iq_site_to_array(data_record):
```

Comment:
As before, the following three methods have a lot of code that is the same. Maybe think about grouping the two IQ methods together as you did before.

Reply:
The tricky part about doing the same thing going backwards is that the bfiq has an extra dimension (number of beams) on the antenna iq data. Also, the size of the first dimension refers to different things in each data set.

Reply:
Disregard, thought of a way around.